wg-node-lifecycle/charter.md

```diff
@@ -0,0 +1,109 @@
+  the new disruption model/evictions.
+- Explore the cloud provider use cases and how they can hook in into the node lifecycle. So that the
+  users can use the same APIs or configurations across the board. 
+- Migrate users of the eviction based kubectl-like drain (kubectl, cluster autoscaler, karpenter,
+  ...) to use the new approach.
+
+
+### Out of scope
+
+- Implementing cloud provider specific logic, the goal is to have high-level API that the providers
+  can use, hook into, or extend.
+- Infrastructure provisioning, deprovisioning solution or physical infrastructure lifecycle
+  management solution.
+
+## Stakeholders
+
+- SIG Apps
+- SIG Architecture
+- SIG Autoscaling
+- SIG CLI
+- SIG Cloud Provider
+- SIG Cluster Lifecycle
+- SIG Node
+- SIG Scheduling
+
+Stakeholders span from multiple SIGs to a broad set of end users,
+public and private cloud providers, Kubernetes distribution providers,
+and cloud provider end-users. Here are some user stories:
+
+- As a cluster admin I want to have a simple interface to initiate a node drain/maintenance without
+  any required manual interventions. I also want to be able to observe the node drain via the API
+  and check on its progress. I also want to be able to discover workloads that are blocking the node
+  drain.
+- To support the new features, node maintenance, scheduler, descheduler, pod autoscaling, kubelet
+  and other actors should use a new eviction API to gracefully remove pods. This should enable new
+  migration strategies that prefer to surge (upscale) pods first rather than downscale them. It
+  should also allow other users/components to monitor pods that are gracefully removed/terminated
+  and provide better behaviour in terms of de/scheduling, scaling and availability. 
+- As an end user, I cannot bear the cost of blue-green upgrades, especially with special hardware
+  accelerators; it's far too expensive. It is more cost-effective to coordinate a drain and then
+  upgrade.
+- As a cloud provider, I need to perform regular maintenance on the hardware in my fleet. Enhancing
+  Kubernetes to help CSPs safely remove hardware will reduce operational costs.
+- Modelling the cost of doing accelerator maintenance in today's world can be massive. And since
+  hardware accelerators tend to need more love and care, having software support to coordinate
+  maintenance will reduce operational costs.
+
+## Deliverables
+
+The WG will coordinate requirement gatherthing and design, eventually leading to
+KEP(s)s and code associated with the ideas.
+
+Area we expect to explore:
+
+- An API to express node drain/maintenance.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4563.
+- An API to solve the problems wrt the API-initiated Eviction API and PDBs.
+  Currently tracked in https://github.com/kubernetes/enhancements/issues/4212.
+- Introduce enhancements across multiple Kubernetes SIGs to add support for the new APIs to solve
+  wide range of issue.
```

Comment:
Core APIs need a reference implementation that execute them and defined semantics with Conformance tests, otherwise this will become implementation specific and non-portable across distros

Reply:
Good idea to mention this explicitly - added below the list.